### PR TITLE
Feat/GitHub actions

### PR DESCRIPTION
**Pull request by thumperward**
### Requirements for Contributing Documentation

* Fill out the template below. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* The pull request must only contribute documentation (for example, markdown files or API docs). To contribute other changes, you must use a different template. You can see all templates at <https://github.com/atom/.github/tree/master/.github/PULL_REQUEST_TEMPLATE>.

### Description of the Change

This PR adds sample configuration for hooking an Atom package repository up to GitHub Actions. It re-uses the script that handles Travis and CircleCI. As GitHub Actions does not presently expose the OS that a task is running on through a common environment variable, the CI configuration sets `TRAVIS_OS_NAME` appropriately so that the handler script needs no modifications.

### Release Notes

- Added sample configuration file for hooking Atom packages up to GitHub Actions for CI/CD

### CODE REVIEW

**Comment by UziTech** [comment]
Is there a way to build on Windows as well?

**Comment by thumperward** [comment]
> Is there a way to build on Windows as well?

I've updated the sample to allow for Windows builds. Note that as the script directly referenced the master branch of this repo, if you want to test it before merging then you'd need to create a local copy which pointed to an existing version of that file (e.g. `https://raw.githubusercontent.com/thumperward/ci/feat/github-actions/build-package.ps1`).

**Comment by UziTech** [comment]
This should also run stable and beta version of Atom on all platforms

**Comment by thumperward** [comment]
> This should also run stable and beta version of Atom on all platforms

added this too, and fixed some schema warnings.

**Comment by UziTech** [comment]
I'm not too familiar with github actions but is there a way that packages creators that want to use this can just specify `use atom/ci@master` to use this action rather than copying the file to their repos?

That should be in the documentation if that is possible.

**Comment by thumperward** [comment]
> I'm not too familiar with github actions but is there a way that packages creators that want to use this can just specify `use atom/ci@master` to use this action rather than copying the file to their repos?

Not directly. Reusable Actions are either JS libraries or Docker containers.

Containerising `build-package.(sh|ps1)` would be a great future step - not only could GitHub then directly use it as an Action with only one line of code, but publishing the container would also simplify its use in other CI systems. However, this is a non-trivial amount of work.

**Comment by UziTech** [comment]
I just created an action that downloads atom and adds `atom` and `apm` to the `PATH`
https://github.com/marketplace/actions/setup-atom

Here is an example action that uses it for testing an apm package:

```yml
name: "CI"
on:
  pull_request:
  push:
    branches:
      - master

env:
  CI: true

jobs:
  Test 👩🏾‍💻:
    strategy:
      matrix:
        os: [ubuntu-latest, macos-latest, windows-latest]
        channel: [stable, beta]
    runs-on: ${{ matrix.os }}
    steps:
    - uses: actions/checkout@v1
    - uses: UziTech/action-setup-atom@v1.0.0
      with:
        channel: ${{ matrix.channel }}
    - name: Atom version
      run: atom -v
    - name: APM version
      run: apm -v
    - name: Install dependencies
      run: apm ci
    - name: Run tests
      run: atom --test spec

  Lint ✨:
    needs: [test]
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
      - name: NPM install
        run: npm ci
      - name: Lint
        run: npm run lint
```

**Comment by UziTech** [comment]
You can see it in action here:
https://github.com/UziTech/terminal-commands/runs/305706692

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    - uses: UziTech/action-setup-atom@v1
+      with:
+        channel: ${{ matrix.channel }}
+    - name: Atom version
+      run: atom -v
+    - name: APM version
+      run: apm -v
+    - name: Install dependencies
+      run: apm install
+    - name: Run coffeelint
```
Comment (UziTech) [inline .github/workflows/main.yml]: If they don't have coffeelint as a dependency this will fail

**Comment by aminya** [comment]
I would like to see this merged.

**Comment by UziTech** [comment]
@aminya you can use github actions without this being merged. Checkout https://github.com/UziTech/terminal-commands/blob/master/.github/workflows/CI.yml for an example.

**Comment by aminya** [comment]
> 
> 
> @aminya you can use github actions without this being merged. Checkout [UziTech/terminal-commands:.github/workflows/CI.yml@`master`](https://github.com/UziTech/terminal-commands/blob/master/.github/workflows/CI.yml) [](/UziTech/terminal-commands/blob/HEAD@{2020-03-16T20:25:48Z}/.github/workflows/CI.yml) for an example.

Thank you very much! I have difficulty loading test packages:
https://github.com/aminya/atom-julia-client/runs/512238925?check_suite_focus=true#step:8:27
Can you help me with it?

**Comment by sadick254** [comment]
@aminya @UziTech Do we still need this change?

**Comment by UziTech** [comment]
It would be nice for this repo to let package owners know how to use GitHub Actions to test Atom packages. I do think this PR needs some changes before being merged.

**Comment by sadick254** [comment]
@UziTech What are some of the change suggestions you have?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v1
+    - uses: UziTech/action-setup-atom@v1
+      with:
+        channel: ${{ matrix.channel }}
+    - name: Atom version
+      run: atom -v
+    - name: APM version
+      run: apm -v
```
Comment (UziTech) [inline .github/workflows/main.yml:23]:
```suggestion
    - uses: UziTech/action-setup-atom@v2
      with:
        version: ${{ matrix.channel }}
```

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    - name: Install dependencies
+      run: apm install
+    - name: Run coffeelint
+      run: ./node_modules/.bin/coffeelint spec
+    - name: Run standard
+      run: ./node_modules/.bin/standard "spec/**/*.js"
```
Comment (UziTech) [inline .github/workflows/main.yml:29]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -60,12 +68,18 @@ the latest Atom release [here](https://atom.io/releases).
...
 The `apm test` command assumes your package is using [Jasmine](http://jasmine.github.io)
 specs. You can run the specs locally using Atom's spec runner UI from the
-_View > Developer > Run Package Specs_ menu or by pressing `cmd-ctrl-alt-p`. You
+`View > Developer > Run Package Specs` menu or by pressing `cmd-ctrl-alt-p`. You
 can run `apm help test` to learn more about that command.
 
-#### Travis CI
+#### GitHub Actions
+
+The CI template uses the [Atom setup Action](UziTech/action-setup-atom@v1) to
```
Comment (UziTech) [inline README.md:76]:
```suggestion
The CI template uses the [Atom setup Action](https://github.com/marketplace/actions/setup-atom) to
```

**Review by UziTech — CHANGES_REQUESTED** [review]

**Comment by UziTech** [comment]
There are the changes I would make to make things simplified. Including v2 of the setup-atom action.

**Review by UziTech — APPROVED** [review]

**Pull request merged**